code/modules/mob/living/carbon/xenomorph/castes/widow/abilities_widow.dm

```diff
@@ -0,0 +1,223 @@
+// ***************************************
+// *********** Web Spit
+// ***************************************
+/datum/action/xeno_action/activable/web_spit
+	name = "Web Spit"
+	ability_name = "Web Spit"
+	mechanics_text = "We spit a stretchy web at our prey"
+	action_icon_state = "toggle_bomb0" // temporary until I get my own icons
+	plasma_cost = 1
+	cooldown_timer = 5 SECONDS
+	keybind_signal = COMSIG_XENOABILITY_WEB_SPIT
+
+/datum/action/xeno_action/activable/web_spit/use_ability(atom/target)
+	var/mob/living/carbon/xenomorph/X = owner
+
+	var/datum/ammo/xeno/acid/web/web_spit = GLOB.ammo_list[/datum/ammo/xeno/acid/web]
+
+	var/obj/projectile/newspit = new /obj/projectile(get_turf(X))
+
+	newspit.generate_bullet(web_spit, web_spit.damage * SPIT_UPGRADE_BONUS(X))
+	newspit.permutated += X
+	newspit.def_zone = X.get_limbzone_target()
+
+	newspit.fire_at(target, X, null, newspit.ammo.max_range)
+
+// ***************************************
+// *********** Burrow
+// ***************************************
+
+/datum/action/xeno_action/activable/burrow
+	name = "Burrow"
+	ability_name = "Burrow"
+	mechanics_text = " Burrow into the ground to hide in plain sight "
+	action_icon_state = "savage_on" //temporary until I get my own icons
+	plasma_cost = 1
+	cooldown_timer = 1 SECONDS
+	keybind_signal = COMSIG_XENOABILITY_BURROW
+
+/datum/action/xeno_action/activable/burrow/use_ability(atom/A)
```

Comment:
if you want spiderling to do the same you need to send a new signal here and on unborrow